### ISSUE TITLE
Starting with 2.1.13 streamed responses not rendered anymore - works fine with 2.1.12

### ISSUE THREAD

**Issue by andnig** [vercel-labs/ai#260]
When using the exact sample application as provided in the [docs](https://vercel.com/blog/introducing-the-vercel-ai-sdk), chat messages are rendered as expected with version ``2.1.12``. However changing the package to ``2.1.13`` does not render the messages anymore.
The requests work fine and responses are coming back to application (seen in chrome devtools), but the messages are not rendered anymore.

```js
'use client'

import { useChat } from 'ai/react'

export default function Chat() {
  const { messages, input, handleInputChange, handleSubmit } = useChat()

  return (
    <div>
      {messages.map(m => (
        <div key={m.id}>
          {m.role}: {m.content}
        </div>
      ))}

      <form onSubmit={handleSubmit}>
        <label>
          Say something...
          <input
            value={input}
            onChange={handleInputChange}
          />
        </label>
      </form>
    </div>
  )
}
```

```js
import { OpenAIStream, StreamingTextResponse } from 'ai'
import { Configuration, OpenAIApi } from 'openai-edge'

// Create an OpenAI API client (that's edge friendly!)
const config = new Configuration({
  apiKey: process.env.OPENAI_API_KEY
})
const openai = new OpenAIApi(config)
 
// IMPORTANT! Set the runtime to edge
export const runtime = 'edge'
 
export async function POST(req: Request) {
  // Extract the `messages` from the body of the request
  const { messages } = await req.json()
 
  // Ask OpenAI for a streaming chat completion given the prompt
  const response = await openai.createChatCompletion({
    model: 'gpt-3.5-turbo',
    stream: true,
    messages
  })
  // Convert the response into a friendly text-stream
  const stream = OpenAIStream(response)
  // Respond with the stream
  return new StreamingTextResponse(stream)
}
```

**Comment by e-roy** [comment]
Here to confirm the same with my results.  Here's my code using langchain

`// api/chat.ts
import { LangChainStream, Message, StreamingTextResponse } from "ai";
import { CallbackManager } from "langchain/callbacks";
import { ChatOpenAI } from "langchain/chat_models/openai";
import {
  AIChatMessage,
  HumanChatMessage,
  SystemChatMessage,
} from "langchain/schema";

export const runtime = "edge";

export async function POST(req: Request) {
  const { messages, model, temperature } = await req.json();

  console.log("messages: ", messages);

  const { stream, handlers } = LangChainStream();

  const llm = new ChatOpenAI({
    modelName: "gpt-3.5-turbo",
    streaming: true,
    callbackManager: CallbackManager.fromHandlers(handlers),
  });

  const chatMessages = (messages as Message[])
    .map((m) => {
      if (m.role === "user") {
        return new HumanChatMessage(m.content);
      }
      if (m.role === "assistant") {
        return new AIChatMessage(m.content);
      }
      if (m.role === "system") {
        return new SystemChatMessage(m.content);
      }
      return undefined; // Default case
    })
    .filter(
      (message): message is Exclude<typeof message, undefined> =>
        message !== undefined
    );

  llm.call(chatMessages).catch(console.error);

  return new StreamingTextResponse(stream);
}
`

What I'm seeing is the message array in the frontend stays empty
`  useEffect(() => {
    console.log("messages", messages);
  }, [messages]);`

This will result in a repeat stream of `messages []`

I am also seeing a response in devtools

**Comment by holdenmatt** [comment]
Yep, also seeing this (seems to affect both the chat and completion APIs).

The examples in the docs here don't work with 2.1.13:
https://sdk.vercel.ai/docs/getting-started
https://sdk.vercel.ai/docs/guides/openai

Works after downgrading to 2.1.12.

**Comment by cuevantn** [comment]
Same issue. The returned value `completion` of `useCompletion` is an empty string:
```js
{ completion: "" }
```

**Comment by mavic111** [comment]
Same issue with huggingface

**Comment by kerkkoh** [comment]
I thought I had gone insane when all of the sudden yesterday everything just didn't work. Glad it wasn't just me.

Here's a quick fix/revert that fixes it:
```ts
const useChatWrapper = (options?: UseChatOptions) => {
  const id = useId();
  const chat = useChat({...(options ?? {}), id})
  return chat;
};

export default function Chat() {
  const { messages, input, handleInputChange, handleSubmit } = useChatWrapper()
  ....
}
```

**Comment by willscottrod** [comment]
> I thought I had gone insane when all of the sudden yesterday everything just didn't work. Glad it wasn't just me.
> 
> Here's a quick fix/revert that fixes it (React/Next.js, but same idea for other frameworks, just provide an id, you can check the old id generation from this commit [fd82961](https://github.com/vercel-labs/ai/commit/fd829612e9c4926c49dc2e207ba709d96d7a0794)):
> 
> ```ts
> import { useId } from 'react';
> import { UseChatOptions, useChat } from 'ai/react'
> 
> const useChatWrapper = (options?: UseChatOptions) => {
>   const id = useId();
>   const chat = useChat({...(options ?? {}), id})
>   return chat;
> };
> 
> export default function Chat() {
>   const { messages, input, handleInputChange, handleSubmit } = useChatWrapper()
>   ....
> }
> ```

This is amazing, this fixed it. I've been trying to fix this for the last 2 hours and was almost losing my mind.

**Comment by anhhtca** [comment]
> quoted...

Amazing ! It's worked.

**Comment by kerkkoh** [comment]
For anyone wanting to actually fix this with a PR and still use nanoid-based ids, change it so that the id-generating function is not called on each render as it currently clearly is. Likely this was missed during testing as I assume they have an id that they provide to `useChat` in the tests.

My fix above is just one way of ensuring there's an id that does not change. You can also just use `useChat({ id: "hello world" })` and it will work just the same, as in:
```ts
import { useChat } from 'ai/react'

export default function Chat() {
  const { messages, input, handleInputChange, handleSubmit } = useChat({ id: "hello world" })
  ....
}
```

**Comment by jaredpalmer** [comment]
This has been resolved. #264 . Sorry about that!

**Comment by kerkkoh** [comment]
@jaredpalmer I'm not sure if there was an actual reason for using nanoid based ids, but if there was, at least in react, just placing the id value in the persistent state to avoid running it on each render should suffice, i.e.
```ts
import {useState} from 'react'
...
const [chatId,] = useState(id || `chat-${nanoid()}`)
```

Again, not sure whether there was a reason for the change, and I am not an expert in svelte nor vue, but I assume they have similar mechanisms to avoid such issues.

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a confirmed regression in existing streaming hooks that broke rendering and state updates after upgrading to version 2.1.13.

### Evidence
- AI SDK 2.1.13 regressed from 2.1.12: streamed responses arrived over the network but `useChat` messages remained empty and `useCompletion` returned an empty string.
- Multiple users reproduced the failure across chat, completion, LangChain, OpenAI, and Hugging Face integrations.
- Providing a stable hook ID worked around the problem, indicating the generated ID was incorrectly changing on each render.
- A maintainer confirmed the issue was resolved in PR #264.
